### PR TITLE
Add support for output directories in fake server.

### PR DESCRIPTION
**Pull request by andusy**
- Adds OutputDirectories to the action result. 
- BuildDir recursively iterates through the directory to build a directory tree, and obtains the digest of that directory tree.

### CODE REVIEW

**Comment by ramymedhat** [comment]
Can you add details as to how you tested this? Also please fix the lint errors.

**Comment by andusy** [comment]
> Can you add details as to how you tested this?

I tested this in reclient by extending the rerun test with the following:
- Added a file/directory to the execroot
- Added a `&fakes.OutputDir` field in `env.Set`
- Checked that OutputDirectoryDigests are present in the resulting metadata and all the new fields are present/updated (TotalOutputBytes, NumOutputDirectories, OutputDirectoryDigests).
- Checked that the digest is the same as the one returned from the local rerun.

**Comment by bozydarsz** [comment]
FYI, regarding failed check builds. looks that https://github.com/googleapis/google-cloud-go/pull/5273 `broke remote-apis-sdks` build. I've merged https://github.com/bazelbuild/remote-apis-sdks/pull/396 to address that. When you rebase your branch to merge in those changes your builds should succeed

**Comment by andusy** [comment]
>  I've merged #396 to address that. 

Oh I didn't see that. Thanks for fixing it!

File: go/pkg/fakes/server.go [hunk 1]
```diff
@@ -279,6 +279,62 @@ func (f *OutputFile) Apply(ac *repb.ActionResult, s *Server, execRoot string) er
...
+	if err != nil {
+		return fmt.Errorf("failed to marshal tree: %v", err)
+	}
+	treeDigest := s.CAS.Put(treeBlob)
+	ac.OutputDirectories = append(ac.OutputDirectories, &repb.OutputDirectory{Path: d.Path, TreeDigest: treeDigest.ToProto()})
+	return nil
+}
+
+// Builds the directory tree by recursively iterating through the directory.
+func BuildDir(path string, s *Server, execRoot string) (root *repb.Directory, childDir []*repb.Directory, err error) {
```
Comment (rubensf) [inline go/pkg/fakes/server.go:307]: Maybe add comments that it looks like `tree.go` `ComputeMerkleTree`, but adapted?

**Review by rubensf — APPROVED** [review]

**Pull request merged**